src/perf/lib/RpsClient.cpp

```diff
@@ -238,8 +241,15 @@ RpsClient::Start(
             QuicSetCurrentThreadProcessorAffinity((uint8_t)ThreadToSetAffinityTo);
     }
 
-
     return QUIC_STATUS_SUCCESS;
+
+Error:
+    for (uint32_t i = 0; i < ConnectionCount; i++) {
+        if (Connections[i] != nullptr) {
+            MsQuic->ConnectionClose(Connections[i]);
+        }
+    }
```

Comment:
Should we just put this in the destructor and remove it from the end of wait as well?

Reply:
That, plus don't even use a temporary variable `Connection`. Put it directly in `Connections[i]`, then we can return on failure and not need to close explicitly.